### PR TITLE
[WB-5841][CLI-765] fix "backend has shutdown"

### PR DESCRIPTION
**Pull request by dannygoldstein**
https://wandb.atlassian.net/browse/WB-5841
https://wandb.atlassian.net/browse/CLI-765

Fixes #1982 

Description
-----------

When you interrupt a running cell in colab, then try to restart it, sometimes it can raise an error that the "wandb backend has shutdown" and require you to restart your interpreter. 

@KyleGoyette traced this to an issue where `interrupt_count` in the internal thread was never reset after resuming the execution of the thread. This means once the interrupt count reached 2, the internal proc could never resume execution.

This PR simply resets this parameter so that the thread can continue executing when the user resumes it

Testing
-------

How was this PR tested?
Tested manually on the colabs provided in https://wandb.atlassian.net/browse/CLI-765 and https://wandb.atlassian.net/browse/WB-5841. Definitely fixed in the case of https://wandb.atlassian.net/browse/WB-5841 and unable to reproduce in https://wandb.atlassian.net/browse/WB-5841. 

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points.
If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------
Fixes "backend has shutdown" error when attempting to rerun colab cells 
------------- END RELEASE NOTES --------------------

### CODE REVIEW

**Comment by dannygoldstein** [comment]
@KyleGoyette i added a test that hits this case, I don't think it will be counted for coverage though since it is running in the notebook.

**Comment by dannygoldstein** [comment]
@KyleGoyette verified that this test fails on master and passes on the PR branch

File: tests/notebooks/repeated_interrupt.ipynb [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": [
+       "1"
+      ]
+     },
+     "metadata": {
+      "tags": []
+     },
+     "execution_count": 5
```
Comment (KyleGoyette) [inline tests/notebooks/repeated_interrupt.ipynb:387]: Why is this 5? Does it even matter?

**Review by KyleGoyette — APPROVED** [review]
LGTM

File: tests/notebooks/repeated_interrupt.ipynb (hunk 1 above)
Comment (dannygoldstein) [reply tests/notebooks/repeated_interrupt.ipynb:387]: doesn't matter, this is just cached from me running it

**Comment by KyleGoyette** [comment]
Coverage is definitely not counting the internal process here. Would be good to get a direct test on the internal_process. But that being difficult, I'd leave it up to Jeff whether this is all right.

**Comment by dmitryduev** [comment]
Tested with the latest version of wandb+service/nexus, and it all seems to work now, thus closing.

**Pull request closed without merging**